src/allocator.rs

```diff
@@ -2,8 +2,10 @@ use crate::err_utils::err;
 use crate::number::{node_from_number, number_from_u8, Number};
 use crate::reduction::EvalErr;
 use chia_bls::{G1Element, G2Element};
+use std::cell::RefCell;
 
-const MAX_NUM_ATOMS: usize = 62500000;
+// subtract 2 to compensate for nil() and one() atoms that we used to always add, unconditionally
+const MAX_NUM_ATOMS: usize = 62500000 - 2;
```

Comment:
What tests do you have for this? Are they just "I expect this test to exceed the limit but minimally" and "I expect this test to hit the limit but just, and finish running" versus "I expect this test to use XXX atoms"? I have concern about this tightening (or worse, loosening) consensus.

Reply:
maybe a more intuitive way of addressing this is to initialize `small_atoms` to 2. This is to compensate for no longer allocating `nil` and `one` in the constructor. I'll make this change.

The tests are:
`test_allocate_atom_limit()`
`test_allocate_small_number_limit()`
`test_allocate_substr_limit()`
`test_allocate_concat_limit()`